DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerBase.py

```diff
@@ -583,7 +583,7 @@ def _getExistingMetadata(self, lfns, connection=False):
       return res
     successful = res['Value']['Successful']
     failed = res['Value']['Failed']
-    for lfn, error in res['Value']['Failed'].items():
+    for lfn, error in res['Value']['Failed'].iteritems():
```

Comment:
This is also going to fail.

Reply:
To expand, this will fail because `failed = res['Value']['Failed']` is not making a copy of the dictionary. So popping in failed is also going to affect `res['Value']['Failed']`

Reply:
Yes, this is clear